tensorflow_estimator/python/estimator/estimator.py

```diff
@@ -1076,8 +1076,14 @@ def _call_input_fn(self, input_fn, mode):
       kwargs['params'] = self.params
     if 'config' in input_fn_args:
       kwargs['config'] = self.config
-    with ops.device('/cpu:0'):
+    if input_context and 'input_context' in input_fn_args:
+      logging.info('`input_context` has been detected in `input_fn`.'
+                   'Placement will be taken over by DistributionStrategy.')
+      kwargs['input_context'] = input_context
```

Comment:
You probably want to add `with ops.device('/cpu:0'):` for this branch as well?

Reply:
Yes, you are right. I thought the device placement should be managed by `InputFunctionIterator` because the `ops.device` also be called inside that function. But that placement seems not always for `cpu`. It is reasonable to add it as well here.

Reply:
Yes, you are right. I thought the device placement should be managed by `InputFunctionIterator` because the `ops.device` also be called inside that function. But that placement seems not always for `cpu`. It is reasonable to add it as well here.